src/handlers/aggregate/global/GlobalFacilityControlAggregate.ts

```diff
@@ -86,16 +86,18 @@ export default class WorldFacilityControlAggregate implements AggregateHandlerIn
         });
 
         try {
-            const row = await this.factory.saveDocument(data);
-            WorldFacilityControlAggregate.logger.debug(`Inserted initial WorldFacilityControlAggregate record for W: ${row.world} | F: ${row.facility}`);
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-ignore
+            const row = await this.factory.model.create(data);
```

Comment:
Feck knows what's wrong with this, something to do with the dynamic data I think.

Reply:
Probably missing the other properties on the model(i.e. nc, tr, vs, totals)

Reply:
Thing is they're there but they get generated via the loop above it, so it's just TS not being intelligent enough to detect it I guess. We could instantiate the properties but it would be messy code, hence why I generated it in a loop.

Reply:
I am not seeing the loop then. I see you create a data variable with an object that contains facility property and that's it.

Reply:
```
factionKeys.forEach((i) => {
            _.mergeWith(
                data,
                {
                    [i]: {
                        captures: 0,
                        defences: 0,
                    },
                },
            );
        });
``` a few lines above

Reply:
Try not to use lodash for everything, just things that can't be done otherwise(e.g. groupBy).

Probably what you want to do is something along the lines of:
```js
(['tr', 'nc', 'vs', 'total'] as (keyof SomeSchema)[]).forEach(k =>{
  data[k] = {}
});
```
And then set data as a type of the `SomeSchema` and that should work.